worlds/_manual/__init__.py

```diff
@@ -166,6 +168,36 @@ def generate_basic(self):
         # then will remove specific item placements below from the overall pool
         self.multiworld.itempool += pool
 
+        # Handle item forbidding
+        locations_with_forbid = [location for location in location_name_to_location.values() if "dont_place_item" in location or "dont_place_item_category" in location]
```

Comment:
I feel like this (both "dont_place_item" and "dont_place_item_category") should follow the same structure as the original `place_item_category` below. In pseudo-code, it'd be something like:

1. Get all location names for `dont_place_*` options, using list comprehension (similar to what you're doing above).
2. Get all locations in the unfilled locations that have matching names to that.
3. Loop over those locations and call `forbid_items_for_player` with the whole list of matched item names (also using list comprehension, like you are below). Doesn't matter if they're in the pool or not.

Should simplify this code some. 😄

Reply:
to use forbid_item(s) you need the real item from the pool not just names if you try to just use the names, for some reason archipelago will ignore those forbidden items its the reason why I check the pool (now with list comprehension)

Reply:
I did it with a dict otherwise im not sure how to pair both list together.
this works for me and uses only list comprehensions and a dict for the manual locations